addinstanceform.php

```diff
@@ -33,17 +36,26 @@
  * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class cachestore_redis_addinstance_form extends cachestore_addinstance_form {
+    /** @var bool */
+    private static $clusteravailable = null;
+
+    protected static function is_cluster_available() {
```

Comment:
Should it not to be in the form class then? Recon you do the same thing in cachestore_redis...